From: Thomas Gleixner <tglx@linutronix.de>
Date: Wed, 6 Jun 2018 00:57:38 +0200
Subject: x86/cpu/AMD: Evaluate smp_num_siblings early
Patch-mainline: Never, SUSE-Xen specific
References: bsc#1089343

To support force disabling of SMT it's required to know the number of
thread siblings early. amd_get_topology() cannot be called before the APIC
driver is selected, so split out the part which initializes
smp_num_siblings and invoke it from amd_early_init().

Signed-off-by: Thomas Gleixner <tglx@linutronix.de>
Acked-by: Ingo Molnar <mingo@kernel.org>
Acked-by: Borislav Petkov <bp@suse.de>
Automatically created from "patches.arch/12-11-x86-cpu-amd-evaluate-smp_num_siblings-early.patch" by xen-port-patches.py

--- a/arch/x86/kernel/cpu/amd.c
+++ b/arch/x86/kernel/cpu/amd.c
@@ -260,8 +260,10 @@ static int __cpuinit nearby_node(int api
 
 static void amd_get_topology_early(struct cpuinfo_x86 *c)
 {
+#ifdef CONFIG_X86_HT
 	if (cpu_has(c, X86_FEATURE_TOPOEXT))
 		smp_num_siblings = ((cpuid_ebx(0x8000001e) >> 8) & 0xff) + 1;
+#endif
 }
 
 /*
